### PR TITLE
Fixed usage of deprecated constructor for 'DateTimeType(Calendar)'

### PR DESCRIPTION
**Pull request by cweitkamp**
- Fixed usage of deprecated constructor for `DateTimeType(Calendar)`

This is a very rough replacement by just converting `Calendar` into `ZonedDateTime` values. I am pretty sure in most cases the logic behind it can be improved even more by doing a deeper refactoring of the code / the origin of theses values.

But it would be nice to get support by the binding developers because it is too error-prone without deeper knowledge.

Related to https://github.com/openhab/openhab-core/pull/1500

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.openhab.binding.logreader/src/main/java/org/openhab/binding/logreader/internal/handler/LogHandler.java [hunk 1]
```diff
@@ -162,7 +163,7 @@ public void fileNotFound() {
     @Override
     public void fileRotated() {
         logger.debug("Log rotated");
-        updateChannelIfLinked(CHANNEL_LOGROTATED, new DateTimeType(Calendar.getInstance()));
+        updateChannelIfLinked(CHANNEL_LOGROTATED, new DateTimeType(ZonedDateTime.now(ZoneId.systemDefault())));
```
Comment (cweitkamp) [inline bundles/org.openhab.binding.logreader/src/main/java/org/openhab/binding/logreader/internal/handler/LogHandler.java:166]:
```suggestion
        updateChannelIfLinked(CHANNEL_LOGROTATED, new DateTimeType(ZonedDateTime.now()));
```

**Comment by bobadair** [comment]
This looks like it should be ok for the Lutron binding, although I have no way to test it for GRXPRG. In both cases it is basically just providing the calculated sunrise/sunset times used by the controller, which is provided without a timezone, so using either the system default timezone or the openHAB default timezone should be appropriate.

I have to say, I don't understand why this call was deprecated. It seems like it would still make sense to have a constructor that takes a Calendar object, as long as it is documented whether it uses the system default ZoneId or the openHAB default ZoneId. Maybe an additional one should be added that accepts a Calendar object and a user-supplied ZoneId.

**Comment by cpmeister** [comment]
Maybe we change the code to use TimeZoneProvider as well.

**Comment by wborn** [comment]
> Maybe an additional one should be added that accepts a Calendar object

By not supporting Calendar anymore we can prevent (unaware) developers from using this outdated API and let them use the preferred `java.time` API instead.

**Pull request merged**